### ISSUE TITLE
Fine tune decoder-only transformers in seq2seq manner

### ISSUE THREAD

**Issue by YerongLi** [huggingface/transformers#27005]
### Feature request

https://github.com/huggingface/transformers/issues/1464
This posts discuss fine-tuning GPT2, 
With GPT2/LLaMA, by default, we need to input the [prompt label] the whole sentence `model([prompt label])` in fine-tuning and caculate the CrossEntropy on the label part, and the model output the model().logits.

**Are there any ways to input the prompt only and do the fine-tuning in the seq2seq manner?** (`model(prompt)`), this way we minimize the loss of log p(y|x).

Get the feature of `model(prompt)` rather than `model([prompt label])` is the whole point.

### Motivation

seq2seq equivalence fine-tuning workflow for decoder-only transformers.

### Your contribution

I could submit PR with discussion as a guidance.

**Comment by ArthurZucker** [comment]
Hey 🤗  thanks for opening an issue! We try to keep the github issues for bugs/feature requests.
Could you ask your question on the [forum](https://discuss.huggingface.co/) instead? I'm sure the community will be of help!

Thanks!
